io/net/Network_virtualization_dlpar.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>
+
+import os
+import shutil
+try:
+    import pxssh
+except ImportError:
+    from pexpect import pxssh
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils.process import CmdError
+from avocado import skipIf, skipUnless
+
+IS_POWER_NV = 'PowerNV' in open('/proc/cpuinfo', 'r').read()
+IS_KVM_GUEST = 'qemu' in open('/proc/cpuinfo', 'r').read()
+
+
+class CommandFailed(Exception):
+    def __init__(self, command, output, exitcode):
+        self.command = command
+        self.output = output
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class NetworkVirtualization(Test):
+    '''
+    Adding and deleting Network Virtualized ports from the vios
+    '''
+    @skipUnless("ppc" in distro.detect().arch,
+                "supported only on Power platform")
```

Comment:
Please add skipIf only once here. There is no need for adding it before `test()` function or any other function as we do not want to execute the `setup()` if test needs to be skipped.
```
@skipIf(IS_POWER_NV or IS_KVM_GUEST, 
        "This test is not supported on KVM guest or PowerNV platform")
```

Reply:
Note that Skip message in the above comment is moved to next line as the previous implementation seemed to have caused unnecessary spaces.
```
 TestSkipError: This test is not supported on KVM         guest or PowerNV platform
```